### DISCUSSION TITLE
Exporting zones for reporting

### DISCUSSION THREAD

**Discussion by allahshukur-ahmadzada** [ngoduykhanh/PowerDNS-Admin#1450] [Q&A] [answered]
Hi, I wonder is there any functionality to export zones from ui? as pdf, file, cvs anything would be suitable for reporting

**Comment by AzorianMatt** [comment, marked as answer]
@allahshukur-ahmadzada there is not currently a feature for this. However, some future work will bring this convenient ability through the use of updated DataTables implementations. Typically, this would be a fairly easy drop-in upgrade, but unfortunately the way it has been implemented currently is quite a mess and not nearly as friendly for doing so.

Once the upgrade is done though, the ability to export via PDF and CSV will be present as a one-click option on the UI.

**Comment by arbaldin** [comment]
@allahshukur-ahmadzada Although we can't export the zone from the UI, it's fairly simple to do it from the PDNS Server shell using pdnsutil:

`pdnsutil list-zone example.com`

This will print the zone as if it was a bind db zone file. I believe you can import the output to excel easily, if I'm not wrong all columns are separated by tabs.
